qemu/tests/migration_multi_host_cancel.py

```diff
@@ -48,39 +53,35 @@ def check_guest(self):
             for vm in self.vms:
                 try:
                     vm = env.get_vm(vm)
+                    stress_kill_cmd = params.get("stress_kill_cmd",
+                                                 "killall -9 stress")
                     session = vm.wait_for_login(timeout=self.login_timeout)
-                    session.sendline("killall -9 cpuflags-test")
+                    session.sendline(stress_kill_cmd)
+                    vm.reboot()
                 except (remote.LoginError, virt_vm.VMError):
                     broken_vms.append(vm)
             if broken_vms:
                 raise error.TestError("VMs %s should work on src"
                                       " host after canceling of"
                                       " migration." % (broken_vms))
-            # Try migration again without cancel.
 
         def migration_scenario(self):
-            srchost = self.params.get("hosts")[0]
-            dsthost = self.params.get("hosts")[1]
-
             def worker(mig_data):
                 vm = mig_data.vms[0]
-                session = vm.wait_for_login(timeout=self.login_timeout)
+                stress_cmd = params.get("stress_cmd")
 
-                cpuflags.install_cpuflags_util_on_vm(test, vm,
-                                                     self.install_path,
-                                                     extra_flags="-msse3 -msse2")
+                session = vm.wait_for_login(timeout=self.login_timeout)
 
-                cmd = ("%s/cpuflags-test --stressmem %d,%d %%" %
-                       (os.path.join(self.install_path, "cpu_flags"),
-                        self.vm_mem * 10, self.vm_mem / 2))
-                logging.debug("Sending command: %s" % (cmd))
-                session.sendline(cmd)
+                logging.info("Load stress in guest before migration.")
+                logging.debug("Sending command: %s" % (stress_cmd))
+                session.sendline(stress_cmd)
 
-            error.context("Migration from %s to %s over protocol %s with high"
-                          " cpu and memory load." %
-                          (self.srchost, self.dsthost, mig_protocol),
+            cancel_delay = params.get("cancel_delay")
+            error.context("Do migration and wait %s seconds, "
+                          "execute 'migrate_cancel'." % cancel_delay,
```

Comment:
1. You want to cancel migration after "cancel_delay" sec, but didn't find delay operation
2. the original error.context looks ok, you can do minor updating and add  description for "cancel" before cancel operation